packages/ui-extensions/docs/surfaces/point-of-sale/mdxExamples/subscription-example/Modal.jsx

```diff
@@ -0,0 +1,67 @@
+import {render} from 'preact';
+import {useState, useEffect} from 'preact/hooks';
+import {fetchSellingPlans} from './FetchSellingPlans';
+
+export default function extension() {
+  render(<Modal />, document.body);
+}
+
+function Modal() {
+  // For this example, we'll just use the first selling plan item
+  // Your app should handle displaying multiple line items with selling plan groups.
+  const sellingPlanItem = shopify.cart.current.value.lineItems.find(
+    (lineItem) => lineItem.hasSellingPlanGroups === true,
+  );
+
+  const [response, setResponse] = useState(undefined);
+
+  useEffect(() => {
+    async function getSellingPlans() {
+      setResponse(await fetchSellingPlans(sellingPlanItem?.variantId));
+    }
+    getSellingPlans();
+  }, [sellingPlanItem]);
+
+  // [START modal.handle-click]
+  const handleClick = (plan) => {
+    shopify.cart.addLineItemSellingPlan({
+      lineItemUuid: sellingPlanItem.uuid,
+      // convert from GID to ID
+      sellingPlanId: Number(plan.id.split('/').pop()),
+      sellingPlanName: plan.name,
+    });
+    window.close();
+  };
+  // [END modal.handle-click]
+
+  // [START modal.display]
+  return (
+    <s-page heading="POS modal">
```

Comment:
Might be good to name this something a bit more contextually relevant, but it's not that important. Feel free to ignore.